File: backend/src/db/seed-synthetic-data.py
#!/usr/bin/env python3

"""
Synthetic Data Seeding Script for B2TW POS
Clears all transaction data and generates realistic test data
Daily sales: 1.5k - 6k
Generates data for the last 30 days
"""

import sqlite3
import uuid
import random
from datetime import datetime, timedelta
from pathlib import Path

# Database path
DB_PATH = Path(__file__).parent.parent.parent.parent / "data" / "pos.db"

# Product catalog with prices (GST-inclusive)
PRODUCTS = [
    # BAO (₹150 with 5% GST)
    {'id': 'prod-001', 'name': 'Chicken Bao', 'price': 150, 'is_beverage': 0},
    {'id': 'prod-002', 'name': 'Soya Bao', 'price': 150, 'is_beverage': 0},
    {'id': 'prod-003', 'name': 'Paneer Bao', 'price': 150, 'is_beverage': 0},
    {'id': 'prod-004', 'name': 'Sabz Bao', 'price': 150, 'is_beverage': 0},
    {'id': 'prod-005', 'name': 'Corn Bao', 'price': 150, 'is_beverage': 0},

    # CHICKEN WINGS (₹180 with 5% GST)
    {'id': 'prod-010', 'name': 'Korean Wings', 'price': 180, 'is_beverage': 0},
    {'id': 'prod-011', 'name': 'Kimchi Wings', 'price': 180, 'is_beverage': 0},
    {'id': 'prod-012', 'name': 'Honey Chilli Chicken', 'price': 180, 'is_beverage': 0},
    {'id': 'prod-013', 'name': 'Sriracha Wings', 'price': 180, 'is_beverage': 0},
    {'id': 'prod-015', 'name': 'Mango Habenaro Wings', 'price': 180, 'is_beverage': 0},

    # KOREAN RAMEN (₹180 with 5% GST)
    {'id': 'prod-006', 'name': 'Kimchi Non Veg Ramen', 'price': 180, 'is_beverage': 0},
    {'id': 'prod-007', 'name': 'Cheese Non Veg Ramen', 'price': 180, 'is_beverage': 0},
    {'id': 'prod-008', 'name': 'Kimchi Veg Ramen', 'price': 180, 'is_beverage': 0},
    {'id': 'prod-009', 'name': 'Cheese Veg Ramen', 'price': 180, 'is_beverage': 0},

    # BEVERAGES
    {'id': 'prod-043', 'name': 'Coke 20', 'price': 20, 'is_beverage': 1},
    {'id': 'prod-044', 'name': 'Sprite 20', 'price': 20, 'is_beverage': 1},
    {'id': 'prod-045', 'name': 'ThumsUp 20', 'price': 20, 'is_beverage': 1},
    {'id': 'prod-034', 'name': 'Mojito Boba', 'price': 175, 'is_beverage': 1},
    {'id': 'prod-035', 'name': 'Ocean Boba', 'price': 175, 'is_beverage': 1},

    # LITE BITES (₹135 with 5% GST)
    {'id': 'prod-020', 'name': 'Chicken Popcorn', 'price': 135, 'is_beverage': 0},
    {'id': 'prod-021', 'name': 'Cheese Pizza Pop', 'price': 135, 'is_beverage': 0},
    {'id': 'prod-022', 'name': 'Veg Spring Roll', 'price': 135, 'is_beverage': 0},
]

CUSTOMER_NAMES = [
    'Raj Kumar', 'Priya Singh', 'Amit Patel', 'Sneha Sharma', 'Vikram Reddy',
    'Anjali Desai', 'Rohit Nair', 'Kavya Menon', 'Arjun Gupta', 'Neha Iyer',
]

PAYMENT_METHODS = ['cash', 'card', 'upi']
STATUSES = ['completed'] * 5 + ['cancelled']  # 83% completed

def calculate_unit_price(menu_price):
    """Calculate pre-tax unit price from GST-inclusive menu price"""
    return round(menu_price / 1.05, 2)

def calculate_gst(unit_price):
    """Calculate 5% GST on unit price"""
    return round(unit_price * 0.05, 2)

def calculate_total_price(unit_price):
    """Calculate total price (unit price + GST) for one unit"""
    gst = calculate_gst(unit_price)
    return round(unit_price + gst, 2)

def get_random_time_for_day(date):
    """Generate a random time within restaurant hours (11 AM - 11 PM)"""
    hour = random.randint(11, 22)
    minute = random.randint(0, 59)
    second = random.randint(0, 59)
    return date.replace(hour=hour, minute=minute, second=second)

def generate_daily_sales_distribution(target_amount):
    """Generate realistic daily sales distribution"""
    return {
        'morning': int(target_amount * 0.05),    # 5% (11-12)
        'lunch': int(target_amount * 0.35),      # 35% (12-3)
        'afternoon': int(target_amount * 0.10),  # 10% (3-5)
        'evening': int(target_amount * 0.40),    # 40% (5-9)
        'night': int(target_amount * 0.10),      # 10% (9-11)
    }

def clear_transaction_data(conn):
    """Clear all transaction data"""
    print('🗑️  Clearing transaction data...')

    tables = [
        'bill_holds',
        'bills',
        'kot_logs',
        'order_items',
        'orders',
        'customers',
    ]

    cursor = conn.cursor()
    for table in tables:
        cursor.execute(f'DELETE FROM {table}')
        print(f'  ✓ Cleared {table}')

    conn.commit()

def seed_products(conn):
    """Insert base products into the database"""
    print('\n🏪 Seeding products...')

    cursor = conn.cursor()

    # Check if products already exist
    cursor.execute('SELECT COUNT(*) FROM products')
    if cursor.fetchone()[0] > 0:
        print('  ✓ Products already exist')
        return

    now = datetime.now().isoformat()

    for idx, product in enumerate(PRODUCTS):
        product_id = str(uuid.uuid4())
        cursor.execute('''
            INSERT INTO products (id, name, category, price, is_beverage, is_active, position, created_at, updated_at)
            VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?)
        ''', (
            product['id'],
            product['name'],
            'Food',
            product['price'],
            product['is_beverage'],
            1,
            idx,
            now,
            now
        ))

    conn.commit()
    print(f'  ✓ Created {len(PRODUCTS)} products')

def generate_customers(conn, count=50):
    """Generate synthetic customers"""
    print(f'\n👥 Generating {count} customers...')

    customers = []
    cursor = conn.cursor()

    for i in range(count):
        phone = f'987654{i:04d}'
        name = f'{CUSTOMER_NAMES[i % len(CUSTOMER_NAMES)]} {i}'
        email = f'customer{i}@example.com'
        customer_id = str(uuid.uuid4())
        now = datetime.now().isoformat()

        cursor.execute('''
            INSERT INTO customers (id, phone, name, email, is_active, created_at, updated_at)
            VALUES (?, ?, ?, ?, ?, ?, ?)
        ''', (customer_id, phone, name, email, 1, now, now))

        customers.append({'id': customer_id, 'phone': phone, 'name': name})

    conn.commit()
    print(f'  ✓ Created {count} customers')
    return customers

def generate_random_order(customers, date):
    """Generate a random order with 1-4 items"""
    customer = random.choice(customers)
    item_count = random.randint(1, 4)
    items = []
    total_amount = 0

    for _ in range(item_count):
        product = random.choice(PRODUCTS)
        quantity = random.randint(1, 3)

        # Calculate prices with correct GST structure
        # menu_price is GST-inclusive, we need to split it
        unit_price = calculate_unit_price(product['price'])  # pre-tax price
        gst_per_unit = calculate_gst(unit_price)  # 5% tax
        total_per_unit = unit_price + gst_per_unit  # pre-tax + tax
        item_total = round(total_per_unit * quantity, 2)  # multiply by quantity

        items.append({
            'product_id': product['id'],
            'product_name': product['name'],
            'quantity': quantity,
            'unit_price': unit_price,
            'total_price': item_total,
        })

        total_amount += item_total

    return {
        'customer_id': customer['id'],
        'customer_phone': customer['phone'],
        'items': items,
        'total_amount': round(total_amount, 2),
        'status': random.choice(STATUSES),
        'payment_method': random.choice(PAYMENT_METHODS),
        'created_at': get_random_time_for_day(date),
    }

def generate_daily_sales_data(conn, customers, date, target_amount):
    """Generate synthetic sales data for a given date"""
    distribution = generate_daily_sales_distribution(target_amount)

    orders = []
    current_amount = 0
    orders_generated = 0

    time_ranges = [
        {'period': 'morning', 'start': 11, 'end': 12, 'amount': distribution['morning']},
        {'period': 'lunch', 'start': 12, 'end': 15, 'amount': distribution['lunch']},
        {'period': 'afternoon', 'start': 15, 'end': 17, 'amount': distribution['afternoon']},
        {'period': 'evening', 'start': 17, 'end': 21, 'amount': distribution['evening']},
        {'period': 'night', 'start': 21, 'end': 23, 'amount': distribution['night']},
    ]

    cursor = conn.cursor()

    for range_info in time_ranges:
        period_amount = 0

        while period_amount < range_info['amount']:
            order = generate_random_order(customers, date)

            # Adjust time to be within range
            order_date = order['created_at']
            hour = random.randint(range_info['start'], range_info['end'])
            order_date = order_date.replace(hour=hour, minute=random.randint(0, 59))
            order['created_at'] = order_date

            orders.append(order)
            period_amount += order['total_amount']
            current_amount += order['total_amount']
            orders_generated += 1

            # Limit orders per period
            period_orders = [o for o in orders if range_info['start'] <= o['created_at'].hour < range_info['end']]
            if len(period_orders) > 30:
                break

    # Insert into database
    for order in orders:
        order_id = str(uuid.uuid4())

        cursor.execute('''
            INSERT INTO orders (id, customer_id, total_amount, status, payment_method, created_at, updated_at)
            VALUES (?, ?, ?, ?, ?, ?, ?)
        ''', (
            order_id,
            order['customer_id'],
            order['total_amount'],
            order['status'],
            order['payment_method'],
            order['created_at'].isoformat(),
            order['created_at'].isoformat()
        ))

        # Insert order items
        for item in order['items']:
            item_id = str(uuid.uuid4())
            cursor.execute('''
                INSERT INTO order_items (id, order_id, product_id, quantity, unit_price, total_price, created_at)
                VALUES (?, ?, ?, ?, ?, ?, ?)
            ''', (
                item_id,
                order_id,
                item['product_id'],
                item['quantity'],
                item['unit_price'],
                item['total_price'],
                order['created_at'].isoformat()
            ))

        # Insert bill if order is completed
        if order['status'] == 'completed':
            bill_id = str(uuid.uuid4())
            bill_number = f"B{str(int(order['created_at'].timestamp()))[-8:]}"

            cursor.execute('''
                INSERT INTO bills (id, order_id, bill_number, customer_id, total_amount, payment_method, status, printed_at, created_at, updated_at)
                VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
            ''', (
                bill_id,
                order_id,
                bill_number,
                order['customer_id'],
                order['total_amount'],
                order['payment_method'],
                'completed',
                order['created_at'].isoformat(),
                order['created_at'].isoformat(),
                order['created_at'].isoformat()
            ))

    conn.commit()

    return {
        'orders_count': orders_generated,
        'total_amount': round(current_amount, 2),
    }

def main():
    """Main seeding function"""
    print('\n╔════════════════════════════════════════════════════════╗')
    print('║     Synthetic Data Seeding for B2TW POS               ║')
    print('╚════════════════════════════════════════════════════════╝\n')

    try:
        # Ensure database directory exists
        DB_PATH.parent.mkdir(parents=True, exist_ok=True)

        # Connect to database
        conn = sqlite3.connect(str(DB_PATH))
        conn.execute('PRAGMA foreign_keys = ON')

        print(f'📦 Connected to database: {DB_PATH}')

        # Initialize schema if needed
        schema_path = Path(__file__).parent / 'schema.sql'
        cursor = conn.cursor()
        cursor.execute("SELECT name FROM sqlite_master WHERE type='table' AND name='orders'")
        if not cursor.fetchone():
            print('\n📋 Initializing database schema...')
            with open(schema_path, 'r') as f:
                schema = f.read()
            cursor.executescript(schema)
            conn.commit()
            print('  ✓ Database schema created')

        # Clear existing data
        clear_transaction_data(conn)

        # Seed products
        seed_products(conn)

        # Generate customers
        customers = generate_customers(conn, 50)

        # Generate sales data for last 30 days
        print('\n📊 Generating 30 days of sales data...')

        today = datetime.now().replace(hour=0, minute=0, second=0, microsecond=0)
        total_revenue = 0
        total_orders = 0

        for days_ago in range(29, -1, -1):
            date = today - timedelta(days=days_ago)
            daily_target = random.randint(1500, 6000)

            day_data = generate_daily_sales_data(conn, customers, date, daily_target)

            total_revenue += day_data['total_amount']
            total_orders += day_data['orders_count']

            date_str = date.strftime('%Y-%m-%d')
            print(f'  {date_str}: {day_data["orders_count"]} orders | ₹{day_data["total_amount"]:.2f}')

        print('\n✅ Seeding complete!')
        print('\n📈 Summary:')
        print(f'  Total Orders: {total_orders}')
        print(f'  Total Revenue: ₹{total_revenue:.2f}')
        print(f'  Avg Daily Revenue: ₹{(total_revenue / 30):.2f}')
        print(f'  Avg Order Value: ₹{(total_revenue / total_orders):.2f}')
        print(f'  Customers: {len(customers)}')

        conn.close()
        print('\n✨ All done! Ready for testing.\n')

    except Exception as e:
        print(f'\n❌ Error seeding data: {e}')
        import traceback
        traceback.print_exc()
        exit(1)

if __name__ == '__main__':
    main()

File: backend/src/db/seed-synthetic-data.js
#!/usr/bin/env node

/**
 * Synthetic Data Seeding Script
 * Clears all transaction data and generates realistic test data
 * Daily sales: 1.5k - 6k
 * Generates data for the last 30 days
 */

import Database from 'better-sqlite3';
import { v4 as uuidv4 } from 'uuid';
import { fileURLToPath } from 'url';
import { dirname, join } from 'path';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

// Product catalog with prices (GST-inclusive)
const PRODUCTS = [
  // BAO (₹150 with 5% GST)
  { id: 'prod-001', name: 'Chicken Bao', price: 150, is_beverage: 0, category: 'Bao' },
  { id: 'prod-002', name: 'Soya Bao', price: 150, is_beverage: 0, category: 'Bao' },
  { id: 'prod-003', name: 'Paneer Bao', price: 150, is_beverage: 0, category: 'Bao' },
  { id: 'prod-004', name: 'Sabz Bao', price: 150, is_beverage: 0, category: 'Bao' },
  { id: 'prod-005', name: 'Corn Bao', price: 150, is_beverage: 0, category: 'Bao' },

  // CHICKEN WINGS (₹180 with 5% GST)
  { id: 'prod-010', name: 'Korean Wings', price: 180, is_beverage: 0, category: 'Chicken Wings' },
  { id: 'prod-011', name: 'Kimchi Wings', price: 180, is_beverage: 0, category: 'Chicken Wings' },
  { id: 'prod-012', name: 'Honey Chilli Chicken', price: 180, is_beverage: 0, category: 'Chicken Wings' },
  { id: 'prod-013', name: 'Sriracha Wings', price: 180, is_beverage: 0, category: 'Chicken Wings' },
  { id: 'prod-015', name: 'Mango Habenaro Wings', price: 180, is_beverage: 0, category: 'Chicken Wings' },

  // KOREAN RAMEN (₹180 with 5% GST)
  { id: 'prod-006', name: 'Kimchi Non Veg Ramen', price: 180, is_beverage: 0, category: 'Korean Ramen' },
  { id: 'prod-007', name: 'Cheese Non Veg Ramen', price: 180, is_beverage: 0, category: 'Korean Ramen' },
  { id: 'prod-008', name: 'Kimchi Veg Ramen', price: 180, is_beverage: 0, category: 'Korean Ramen' },
  { id: 'prod-009', name: 'Cheese Veg Ramen', price: 180, is_beverage: 0, category: 'Korean Ramen' },

  // BEVERAGES
  { id: 'prod-043', name: 'Coke 20', price: 20, is_beverage: 1, category: 'Beverages' },
  { id: 'prod-044', name: 'Sprite 20', price: 20, is_beverage: 1, category: 'Beverages' },
  { id: 'prod-045', name: 'ThumsUp 20', price: 20, is_beverage: 1, category: 'Beverages' },
  { id: 'prod-034', name: 'Mojito Boba', price: 175, is_beverage: 1, category: 'Beverages' },
  { id: 'prod-035', name: 'Ocean Boba', price: 175, is_beverage: 1, category: 'Beverages' },

  // LITE BITES (₹135 with 5% GST)
  { id: 'prod-020', name: 'Chicken Popcorn', price: 135, is_beverage: 0, category: 'Lite Bites' },
  { id: 'prod-021', name: 'Cheese Pizza Pop', price: 135, is_beverage: 0, category: 'Lite Bites' },
  { id: 'prod-022', name: 'Veg Spring Roll', price: 135, is_beverage: 0, category: 'Lite Bites' },
];

// Sample customer phone numbers (10 digit Indian format)
const CUSTOMER_PHONES = [
  '9876543210', '9876543211', '9876543212', '9876543213', '9876543214',
  '9876543215', '9876543216', '9876543217', '9876543218', '9876543219',
  '9123456789', '9123456790', '9123456791', '9123456792', '9123456793',
  '9876543220', '9876543221', '9876543222', '9876543223', '9876543224',
];

const CUSTOMER_NAMES = [
  'Raj Kumar', 'Priya Singh', 'Amit Patel', 'Sneha Sharma', 'Vikram Reddy',
  'Anjali Desai', 'Rohit Nair', 'Kavya Menon', 'Arjun Gupta', 'Neha Iyer',
];

const PAYMENT_METHODS = ['cash', 'card', 'upi'];
const STATUSES = ['completed', 'completed', 'completed', 'completed', 'completed', 'cancelled']; // 83% completed

/**
 * Generate a random number within a range
 */
function randomBetween(min, max) {
  return Math.floor(Math.random() * (max - min + 1)) + min;
}

/**
 * Generate a random datetime for a given day
 */
function getRandomTimeForDay(date) {
  const newDate = new Date(date);
  // Restaurant likely open 11 AM to 11 PM
  const hour = randomBetween(11, 22);
  const minute = randomBetween(0, 59);
  const second = randomBetween(0, 59);
  newDate.setHours(hour, minute, second);
  return newDate;
}

/**
 * Generate realistic daily sales distribution
 * More orders during lunch (12-2 PM) and dinner (6-9 PM)
 */
function generateDailySalesDistribution(targetAmount) {
  const distribution = {
    morning: 0.05,   // 5% (11-12)
    lunch: 0.35,     // 35% (12-3)
    afternoon: 0.10, // 10% (3-5)
    evening: 0.40,   // 40% (5-9)
    night: 0.10      // 10% (9-11)
  };

  return {
    morning: Math.round(targetAmount * distribution.morning),
    lunch: Math.round(targetAmount * distribution.lunch),
    afternoon: Math.round(targetAmount * distribution.afternoon),
    evening: Math.round(targetAmount * distribution.evening),
    night: Math.round(targetAmount * distribution.night),
  };
}

/**
 * Clear all transaction data
 */
function clearTransactionData(db) {
  console.log('🗑️  Clearing transaction data...');

  const tables = [
    'bill_holds',
    'bills',
    'kot_logs',
    'order_items',
    'orders',
    'customers',
  ];

  for (const table of tables) {
    db.prepare(`DELETE FROM ${table}`).run();
    console.log(`  ✓ Cleared ${table}`);
  }
}

/**
 * Generate synthetic customers
 */
function generateCustomers(db, count = 50) {
  console.log(`\n👥 Generating ${count} customers...`);

  const customers = [];
  for (let i = 0; i < count; i++) {
    const phone = CUSTOMER_PHONES[i % CUSTOMER_PHONES.length].slice(0, -1) + i;
    const name = CUSTOMER_NAMES[i % CUSTOMER_NAMES.length] + ' ' + i;

    const customer = {
      id: uuidv4(),
      phone,
      name,
      email: `customer${i}@example.com`,
      is_active: 1,
      created_at: new Date().toISOString(),
      updated_at: new Date().toISOString(),
    };

    db.prepare(`
      INSERT INTO customers (id, phone, name, email, is_active, created_at, updated_at)
      VALUES (?, ?, ?, ?, ?, ?, ?)
    `).run(customer.id, customer.phone, customer.name, customer.email, customer.is_active, customer.created_at, customer.updated_at);

    customers.push(customer);
  }

  console.log(`  ✓ Created ${count} customers`);
  return customers;
}

/**
 * Generate a random order with 1-4 items
 */
function generateRandomOrder(customers, date) {
  const customer = customers[randomBetween(0, customers.length - 1)];
  const itemCount = randomBetween(1, 4);
  const items = [];
  let totalAmount = 0;

  for (let i = 0; i < itemCount; i++) {
    const product = PRODUCTS[randomBetween(0, PRODUCTS.length - 1)];
    const quantity = randomBetween(1, 3);
    const itemTotal = product.price * quantity;

    items.push({
      product_id: product.id,
      product_name: product.name,
      quantity,
      unit_price: product.price,
      total_price: itemTotal,
    });

    totalAmount += itemTotal;
  }

  return {
    customer_id: customer.id,
    customer_phone: customer.phone,
    items,
    totalAmount: Math.round(totalAmount * 100) / 100,
    status: STATUSES[randomBetween(0, STATUSES.length - 1)],
    payment_method: PAYMENT_METHODS[randomBetween(0, PAYMENT_METHODS.length - 1)],
    created_at: getRandomTimeForDay(date),
  };
}

/**
 * Generate synthetic sales data for a given date
 */
function generateDailySalesData(db, customers, date, targetAmount) {
  const distribution = generateDailySalesDistribution(targetAmount);
  const baseDate = new Date(date);

  const orders = [];
  let currentAmount = 0;
  let ordersGenerated = 0;

  // Generate orders across time periods
  const timeRanges = [
    { period: 'morning', start: 11, end: 12, amount: distribution.morning },
    { period: 'lunch', start: 12, end: 15, amount: distribution.lunch },
    { period: 'afternoon', start: 15, end: 17, amount: distribution.afternoon },
    { period: 'evening', start: 17, end: 21, amount: distribution.evening },
    { period: 'night', start: 21, end: 23, amount: distribution.night },
  ];

  for (const range of timeRanges) {
    let periodAmount = 0;

    while (periodAmount < range.amount) {
      const order = generateRandomOrder(customers, date);

      // Update the created_at time to be within the time range
      const orderDate = new Date(baseDate);
      const hour = randomBetween(range.start, range.end);
      const minute = randomBetween(0, 59);
      orderDate.setHours(hour, minute, randomBetween(0, 59));
      order.created_at = orderDate;

      orders.push(order);
      periodAmount += order.totalAmount;
      currentAmount += order.totalAmount;
      ordersGenerated++;

      // Limit orders per period
      if (orders.filter(o => {
        const h = new Date(o.created_at).getHours();
        return h >= range.start && h < range.end;
      }).length > 30) break;
    }
  }

  // Insert orders and items into database
  for (const order of orders) {
    const orderId = uuidv4();

    db.prepare(`
      INSERT INTO orders (id, customer_id, total_amount, status, payment_method, created_at, updated_at)
      VALUES (?, ?, ?, ?, ?, ?, ?)
    `).run(
      orderId,
      order.customer_id,
      order.totalAmount,
      order.status,
      order.payment_method,
      order.created_at.toISOString(),
      order.created_at.toISOString()
    );

    // Insert order items
    for (const item of order.items) {
      const itemId = uuidv4();
      db.prepare(`
        INSERT INTO order_items (id, order_id, product_id, quantity, unit_price, total_price, created_at)
        VALUES (?, ?, ?, ?, ?, ?, ?)
      `).run(
        itemId,
        orderId,
        item.product_id,
        item.quantity,
        item.unit_price,
        item.total_price,
        order.created_at.toISOString()
      );
    }

    // Insert corresponding bill
    if (order.status === 'completed') {
      const billId = uuidv4();
      const billNumber = `B${order.created_at.getTime().toString().slice(-8)}`;

      db.prepare(`
        INSERT INTO bills (id, order_id, bill_number, customer_id, total_amount, payment_method, status, printed_at, created_at, updated_at)
        VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
      `).run(
        billId,
        orderId,
        billNumber,
        order.customer_id,
        order.totalAmount,
        order.payment_method,
        'completed',
        order.created_at.toISOString(),
        order.created_at.toISOString(),
        order.created_at.toISOString()
      );
    }
  }

  return {
    ordersCount: ordersGenerated,
    totalAmount: Math.round(currentAmount * 100) / 100,
  };
}

/**
 * Main seeding function
 */
function seedSyntheticData() {
  console.log('\n╔════════════════════════════════════════════════════════╗');
  console.log('║     Synthetic Data Seeding for B2TW POS               ║');
  console.log('╚════════════════════════════════════════════════════════╝\n');

  try {
    // Connect to database
    const dbPath = join(__dirname, '../../..', 'db', 'pos.sqlite');
    const db = new Database(dbPath);
    db.pragma('foreign_keys = ON');

    console.log(`📦 Connected to database: ${dbPath}`);

    // Clear existing data
    clearTransactionData(db);

    // Generate customers
    const customers = generateCustomers(db, 50);

    // Generate sales data for the last 30 days
    console.log('\n📊 Generating 30 days of sales data...');

    const today = new Date();
    let totalRevenue = 0;
    let totalOrders = 0;
    const dailyBreakdown = [];

    for (let daysAgo = 29; daysAgo >= 0; daysAgo--) {
      const date = new Date(today);
      date.setDate(date.getDate() - daysAgo);

      // Generate random daily sales between 1500 and 6000
      const dailyTarget = randomBetween(1500, 6000);

      const dayData = generateDailySalesData(db, customers, date, dailyTarget);

      totalRevenue += dayData.totalAmount;
      totalOrders += dayData.ordersCount;

      const dateStr = date.toISOString().split('T')[0];
      dailyBreakdown.push({
        date: dateStr,
        orders: dayData.ordersCount,
        revenue: dayData.totalAmount,
      });

      console.log(`  ${dateStr}: ${dayData.ordersCount} orders | ₹${dayData.totalAmount.toFixed(2)}`);
    }

    console.log('\n✅ Seeding complete!');
    console.log('\n📈 Summary:');
    console.log(`  Total Orders: ${totalOrders}`);
    console.log(`  Total Revenue: ₹${totalRevenue.toFixed(2)}`);
    console.log(`  Avg Daily Revenue: ₹${(totalRevenue / 30).toFixed(2)}`);
    console.log(`  Avg Order Value: ₹${(totalRevenue / totalOrders).toFixed(2)}`);
    console.log(`\n  Customers: ${customers.length}`);

    db.close();
    console.log('\n✨ All done! Ready for testing.\n');

  } catch (error) {
    console.error('\n❌ Error seeding data:', error.message);
    process.exit(1);
  }
}

// Run the seeding
seedSyntheticData();
